flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1693,35 +1659,85 @@ static mlir::Value createExtremumCompare(mlir::Location loc,
 // VERIFY 
 fir::ExtendedValue
 IntrinsicLibrary::genVerify(mlir::Type resultType,
-                            llvm::ArrayRef<fir::ExtendedValue> args) {
+                                llvm::ArrayRef<fir::ExtendedValue> args) {
+
+  return IntrinsicLibrary::genScanVerify(resultType, args, false);
+}
+
+// Process Scan/Verify intrinsic
+fir::ExtendedValue
+IntrinsicLibrary::genScanVerify(mlir::Type resultType,
+                                llvm::ArrayRef<fir::ExtendedValue> args,
+                                bool isScan) {
```

Comment:
templates might be a more consistent option for code sharing instead of an extra `bool` argument.